### PR TITLE
Add support for Zigbee fota with XiP & new Zigbee weather station + SMP server application

### PR DESCRIPTION
**Pull request by sebastiandraus**

### CODE REVIEW

File: modules/mcuboot/CMakeLists.txt [hunk 1]
```diff
@@ -790,21 +790,39 @@ if (CONFIG_DFU_MULTI_IMAGE_PACKAGE_BUILD)
...
 
-  if (CONFIG_DFU_MULTI_IMAGE_PACKAGE_NET)
-    list(APPEND dfu_multi_image_ids 1)
-    list(APPEND dfu_multi_image_paths "${PROJECT_BINARY_DIR}/${net_core_binary_name}")
-    list(APPEND dfu_multi_image_targets net_core_app_sign_target)
+    if (CONFIG_DFU_MULTI_IMAGE_PACKAGE_APP)
+      list(APPEND dfu_multi_image_ids 0)
+      list(APPEND dfu_multi_image_paths "${PROJECT_BINARY_DIR}/internal_flash_update.bin")
+      list(APPEND dfu_multi_image_targets mcuboot_sign_target)
+    endif()
```
Comment (nordicjm) [inline modules/mcuboot/CMakeLists.txt:803]: QSPI part should be moved inside this if()

File: zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf [hunk 2]
```diff
@@ -51,3 +51,45 @@ CONFIG_BUILD_NO_GAP_FILL=y
...
+#######################
+# Zigbee FOTA overlay #
+#######################
+
+# Enable Zigbee FOTA library
+CONFIG_ZIGBEE_FOTA=y
+CONFIG_ZIGBEE_FOTA_HW_VERSION=52
+CONFIG_ZIGBEE_FOTA_ENDPOINT=5
+CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION="1.2.3"
+# CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION="1.2.4"
```
Comment (nordicjm) [inline zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf]: Duplicate line can go

File: zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf [hunk 3]
```diff
@@ -51,3 +51,45 @@ CONFIG_BUILD_NO_GAP_FILL=y
...
+CONFIG_IMG_ERASE_PROGRESSIVELY=y
+CONFIG_ZIGBEE_FOTA_PROGRESS_EVT=y
+
+# Increase the number of RX buffers
+CONFIG_NRF_802154_RX_BUFFERS=32
+
+
+#######################
+# Zigbee FOTA board overlay #
+#######################
```
Comment (nordicjm) [inline zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf]: Header size not aligned

File: zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj_release.conf [hunk 4]
```diff
@@ -47,3 +47,45 @@ CONFIG_BUILD_NO_GAP_FILL=y
...
+#######################
+# Zigbee FOTA overlay #
+#######################
+
+# Enable Zigbee FOTA library
+CONFIG_ZIGBEE_FOTA=y
+CONFIG_ZIGBEE_FOTA_HW_VERSION=52
+CONFIG_ZIGBEE_FOTA_ENDPOINT=5
+CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION="1.2.3"
+# CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION="1.2.4"
```
Comment (nordicjm) [inline zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj_release.conf]: As above

File: zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj_release.conf [hunk 5]
```diff
@@ -47,3 +47,45 @@ CONFIG_BUILD_NO_GAP_FILL=y
...
+CONFIG_IMG_ERASE_PROGRESSIVELY=y
+CONFIG_ZIGBEE_FOTA_PROGRESS_EVT=y
+
+# Increase the number of RX buffers
+CONFIG_NRF_802154_RX_BUFFERS=32
+
+
+#######################
+# Zigbee FOTA board overlay #
+#######################
```
Comment (nordicjm) [inline zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj_release.conf]: As above

File: zigbee_weather_station_mcumgr/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/sensor.c
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/weather_station.c
+                     LOCATION RAM_DATA)
+
+# zephyr_code_relocate(LIBRARY drivers__ieee802154 LOCATION EXTFLASH_TEXT NOCOPY)
+zephyr_code_relocate(LIBRARY drivers__sensor__adxl362 LOCATION EXTFLASH_TEXT NOCOPY)
+zephyr_code_relocate(LIBRARY drivers__sensor__bme680 LOCATION EXTFLASH_TEXT NOCOPY)
+zephyr_code_relocate(LIBRARY drivers__sensor__bmm150 LOCATION EXTFLASH_TEXT NOCOPY)
+zephyr_code_relocate(LIBRARY ..__nrf__lib__dk_buttons_and_leds LOCATION EXTFLASH_TEXT NOCOPY)
+# zephyr_code_relocate(LIBRARY subsys__net LOCATION EXTFLASH_TEXT NOCOPY)
```
Comment (nordicjm) [inline zigbee_weather_station_mcumgr/CMakeLists.txt]: Commented line can go

File: zigbee_weather_station_mcumgr/README.rst [hunk 7]
```diff
@@ -0,0 +1,541 @@
...
+  For example:
+
+  .. code-block:: console
+
+     [00:20:06.458,801] <inf> app: Attributes T:      2676      P:       967       H:      2317
+
+Building and running
+********************
+
+.. |application path| replace:: :file:`applications/zigbee_weather_station`
```
Comment (nordicjm) [inline zigbee_weather_station_mcumgr/README.rst]: Paths should probably be updated

File: zigbee_weather_station_mcumgr/pm_static.yml [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+  device: MX25R64
+  end_address: 0x120000
+  region: external_flash
+  size: 0x40000
+mcuboot_primary_2:
+  address: 0x120000
+  device: MX25R64
+  end_address: 0x160000
+  region: external_flash
+  size: 0x40000
```
Comment (nordicjm) [inline zigbee_weather_station_mcumgr/pm_static.yml:76]: Just noticed that this is not in sync with the linker file here on in the one it was copied from

File: modules/mcuboot/CMakeLists.txt [hunk 9]
```diff
@@ -790,21 +790,39 @@ if (CONFIG_DFU_MULTI_IMAGE_PACKAGE_BUILD)
...
+    if (CONFIG_DFU_MULTI_IMAGE_PACKAGE_NET)
+      list(APPEND dfu_multi_image_ids 1)
+      list(APPEND dfu_multi_image_paths "${PROJECT_BINARY_DIR}/${net_core_binary_name}")
+      list(APPEND dfu_multi_image_targets net_core_app_sign_target)
+    endif()
+
+    list(APPEND dfu_multi_image_ids 2)
+    list(APPEND dfu_multi_image_paths "${PROJECT_BINARY_DIR}/qspi_flash_update.bin")
+
+
```
Comment (nordicjm) [inline modules/mcuboot/CMakeLists.txt]: Excess whitespace can go

**Review by nordicjm — COMMENTED** [review]
Comments on this PR are still outstanding

**Comment by sebastiandraus** [comment]
Rebased at newest main branch in ncs-playground and removed pointing to PR in nrf repo so this ISN'T WORKING right now.
Aligning nrf repo revision in progress

**Pull request closed without merging**